sql/core/src/main/scala/org/apache/spark/sql/execution/python/BatchEvalPythonExec.scala

```diff
@@ -109,3 +78,43 @@ case class BatchEvalPythonExec(udfs: Seq[PythonUDF], resultAttrs: Seq[Attribute]
   override protected def withNewChildInternal(newChild: SparkPlan): BatchEvalPythonExec =
     copy(child = newChild)
 }
+
+object BatchEvalPythonExec {
```

Comment:
This UDTF is very similar with `MapInPandasExec` FWIW.

Reply:
If so, another approach maybe rewriting UTTF to a `PandasMapIterFunction`?

Reply:
Thanks for the pointer! I will look into this when implementing arrow-based eval for UDTFs.